extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1262,5 +1262,58 @@ describes.realWin('amp-story', {
                 toggleExperiment(win, 'amp-story-branching', false);
               });
         });
+
+    it('should navigate back to the correct previous page after goToPage',
+        () => {
+          toggleExperiment(win, 'amp-story-branching', true);
+          createPages(story.element, 4,
+              ['cover', 'page-1', 'page-2', 'page-3']);
+          story.buildCallback();
+          return story.layoutCallback()
+              .then(() => {
+                story.element.setAttribute('id', 'story');
+
+                const actionButton = createElementWithAttributes(
+                    win.document,
+                    'button',
+                    {'id': 'actionButton',
+                      'on': 'tap:story.goToPage(id=page-2)'});
+                element.querySelector('#cover').appendChild(actionButton);
+                // Click on the actionButton to trigger the goToPage action.
+                actionButton.click();
+
+                //Move backwards
+                story.activePage_.element.dispatchEvent(
+                    new MouseEvent('click', {clientX: 0}));
+                expect(story.activePage_.element.id).to.equal('cover');
+                toggleExperiment(win, 'amp-story-branching', false);
+              });
+        });
+
+    it('should navigate back to the correct previous page after advance-to',
+        () => {
+          toggleExperiment(win, 'amp-story-branching', true);
+          createPages(
+              story.element, 4, ['cover', 'page-1', 'page-2', 'page-3']);
+          story.buildCallback();
+          return story.layoutCallback()
+              .then(() => {
+                story.getPageById('cover')
+                    .element.setAttribute('advance-to', 'page-3');
+
+                expect(story.activePage_.element.id).to.equal('cover');
+                // Move forwards
+                story.activePage_.element.dispatchEvent(
+                    new MouseEvent('click', {clientX: 200}));
+                expect(story.activePage_.element.id).to.equal('page-3');
+
+                // Move backwards
+                story.activePage_.element.dispatchEvent(
+                    new MouseEvent('click', {clientX: 0}));
+                expect(story.activePage_.element.id).to.equal('cover');
+
+                toggleExperiment(win, 'amp-story-branching', false);
```

Comment:
Ideally, a test would only assert one thing, like moving forwards, or moving backwards. Could you please split this test in two tests? It's going to create duplicated code, but in the context of tests it'd actually help debugging, as you'd know right away if you broke forward or backward navigation.

Reply:
This makes sense. I went ahead and removed the expects confirms the correct forwards movement. The advance-to PR itself should already have a test that tests this.